Parser bug caused by some characters

After entering the code below and adding any other character at the marked location, an exception is thrown (see below).

**Code snippet**
```js
const K : string = § // add any other character after '§'
```

**Exception**
```
ERROR - java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at java.base/java.util.concurrent.FutureTask.report(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.get(Unknown Source)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ContentAssistContextFactory.doCreateContexts(ContentAssistContextFactory.java:181)
	at org.eclipse.n4js.ide.editor.contentassist.N4JSContentAssistContextFactory.doCreateContexts(N4JSContentAssistContextFactory.java:90)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ContentAssistContextFactory.create(ContentAssistContextFactory.java:131)
	at org.eclipse.n4js.xtext.ide.server.contentassist.XContentAssistService.createProposals(XContentAssistService.java:95)
	at org.eclipse.n4js.ide.editor.contentassist.N4JSContentAssistService.createProposals(N4JSContentAssistService.java:35)
	at org.eclipse.n4js.xtext.ide.server.contentassist.XContentAssistService.createCompletionList(XContentAssistService.java:70)
	at org.eclipse.n4js.xtext.ide.server.TextDocumentFrontend.completion(TextDocumentFrontend.java:184)
	at org.eclipse.n4js.ide.server.N4JSTextDocumentFrontend.completion(N4JSTextDocumentFrontend.java:83)
	at org.eclipse.n4js.xtext.ide.server.TextDocumentFrontend.lambda$0(TextDocumentFrontend.java:167)
	at org.eclipse.n4js.xtext.ide.server.ResourceTaskManager.lambda$5(ResourceTaskManager.java:332)
	at org.eclipse.n4js.ide.server.concurrent.N4JSQueuedExecutorService.lambda$0(N4JSQueuedExecutorService.java:35)
	at org.eclipse.n4js.xtext.ide.server.QueuedExecutorService$QueuedTask.run(QueuedExecutorService.java:124)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.xtext.ide.editor.contentassist.antlr.BaseContentAssistParser.getFollowElements(BaseContentAssistParser.java:246)
	at org.eclipse.n4js.ide.editor.contentassist.CustomN4JSParser.doGetFollowElements(CustomN4JSParser.java:483)
	at org.eclipse.n4js.ide.editor.contentassist.CustomN4JSParser.collectFollowElements(CustomN4JSParser.java:460)
	at org.eclipse.n4js.ide.editor.contentassist.CustomN4JSParser.getFollowElements(CustomN4JSParser.java:234)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.PartialContentAssistContextFactory.parseFollowElements(PartialContentAssistContextFactory.java:46)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.PartialContentAssistContextFactory.handleLastCompleteNodeAsPartOfDatatypeNode(PartialContentAssistContextFactory.java:52)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ContentAssistContextFactory$1.call(ContentAssistContextFactory.java:149)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ContentAssistContextFactory$1.call(ContentAssistContextFactory.java:146)
	... 4 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.BaseContentAssistParser.getFollowElements(BaseContentAssistParser.java:172)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.BaseContentAssistParser.getFollowElements(BaseContentAssistParser.java:242)
	... 11 more
Caused by: java.lang.IllegalArgumentException: §
	at org.eclipse.n4js.ide.editor.contentassist.ContentAssistTokenTypeMapper.getInternalTokenType(ContentAssistTokenTypeMapper.java:66)
	at org.eclipse.n4js.ide.editor.contentassist.NodeModelTokenSource.nextToken(NodeModelTokenSource.java:218)
	at org.eclipse.n4js.ide.editor.contentassist.NodeModelTokenSource.processHiddenToken(NodeModelTokenSource.java:227)
	at org.eclipse.n4js.ide.editor.contentassist.NodeModelTokenSource.nextToken(NodeModelTokenSource.java:216)
	at org.antlr.runtime.CommonTokenStream.fillBuffer(CommonTokenStream.java:119)
	at org.antlr.runtime.CommonTokenStream.LT(CommonTokenStream.java:238)
	at org.eclipse.xtext.parser.antlr.XtextTokenStream.LT(XtextTokenStream.java:289)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ObservableXtextTokenStream.LA(ObservableXtextTokenStream.java:46)
	at org.eclipse.n4js.ide.contentassist.antlr.internal.InternalN4JSParser.rule__Script__Group__1__Impl(InternalN4JSParser.java:23897)
	at org.eclipse.n4js.ide.contentassist.antlr.internal.InternalN4JSParser.rule__Script__Group__1(InternalN4JSParser.java:23887)
	at org.eclipse.n4js.ide.contentassist.antlr.internal.InternalN4JSParser.rule__Script__Group__0(InternalN4JSParser.java:23868)
	... 17 more
ERROR - java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at java.base/java.util.concurrent.FutureTask.report(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.get(Unknown Source)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ContentAssistContextFactory.doCreateContexts(ContentAssistContextFactory.java:181)
	at org.eclipse.n4js.ide.editor.contentassist.N4JSContentAssistContextFactory.doCreateContexts(N4JSContentAssistContextFactory.java:90)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ContentAssistContextFactory.create(ContentAssistContextFactory.java:131)
	at org.eclipse.n4js.xtext.ide.server.contentassist.XContentAssistService.createProposals(XContentAssistService.java:95)
	at org.eclipse.n4js.ide.editor.contentassist.N4JSContentAssistService.createProposals(N4JSContentAssistService.java:35)
	at org.eclipse.n4js.xtext.ide.server.contentassist.XContentAssistService.createCompletionList(XContentAssistService.java:70)
	at org.eclipse.n4js.xtext.ide.server.TextDocumentFrontend.completion(TextDocumentFrontend.java:184)
	at org.eclipse.n4js.ide.server.N4JSTextDocumentFrontend.completion(N4JSTextDocumentFrontend.java:83)
	at org.eclipse.n4js.xtext.ide.server.TextDocumentFrontend.lambda$0(TextDocumentFrontend.java:167)
	at org.eclipse.n4js.xtext.ide.server.ResourceTaskManager.lambda$5(ResourceTaskManager.java:332)
	at org.eclipse.n4js.ide.server.concurrent.N4JSQueuedExecutorService.lambda$0(N4JSQueuedExecutorService.java:35)
	at org.eclipse.n4js.xtext.ide.server.QueuedExecutorService$QueuedTask.run(QueuedExecutorService.java:124)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.xtext.ide.editor.contentassist.antlr.BaseContentAssistParser.getFollowElements(BaseContentAssistParser.java:246)
	at org.eclipse.n4js.ide.editor.contentassist.CustomN4JSParser.doGetFollowElements(CustomN4JSParser.java:483)
	at org.eclipse.n4js.ide.editor.contentassist.CustomN4JSParser.collectFollowElements(CustomN4JSParser.java:460)
	at org.eclipse.n4js.ide.editor.contentassist.CustomN4JSParser.getFollowElements(CustomN4JSParser.java:234)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.PartialContentAssistContextFactory.parseFollowElements(PartialContentAssistContextFactory.java:46)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.PartialContentAssistContextFactory.createContextsForLastCompleteNode(PartialContentAssistContextFactory.java:41)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ContentAssistContextFactory.handleLastCompleteNodeIsPartOfLookahead(ContentAssistContextFactory.java:220)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ContentAssistContextFactory$3.call(ContentAssistContextFactory.java:172)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ContentAssistContextFactory$3.call(ContentAssistContextFactory.java:169)
	... 4 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.BaseContentAssistParser.getFollowElements(BaseContentAssistParser.java:172)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.BaseContentAssistParser.getFollowElements(BaseContentAssistParser.java:242)
	... 12 more
Caused by: java.lang.IllegalArgumentException: §
	at org.eclipse.n4js.ide.editor.contentassist.ContentAssistTokenTypeMapper.getInternalTokenType(ContentAssistTokenTypeMapper.java:66)
	at org.eclipse.n4js.ide.editor.contentassist.NodeModelTokenSource.nextToken(NodeModelTokenSource.java:218)
	at org.eclipse.n4js.ide.editor.contentassist.NodeModelTokenSource.processHiddenToken(NodeModelTokenSource.java:227)
	at org.eclipse.n4js.ide.editor.contentassist.NodeModelTokenSource.nextToken(NodeModelTokenSource.java:216)
	at org.antlr.runtime.CommonTokenStream.fillBuffer(CommonTokenStream.java:119)
	at org.antlr.runtime.CommonTokenStream.LT(CommonTokenStream.java:238)
	at org.eclipse.xtext.parser.antlr.XtextTokenStream.LT(XtextTokenStream.java:289)
	at org.eclipse.xtext.ide.editor.contentassist.antlr.ObservableXtextTokenStream.LA(ObservableXtextTokenStream.java:46)
	at org.eclipse.n4js.ide.contentassist.antlr.internal.InternalN4JSParser.rule__Script__Group__1__Impl(InternalN4JSParser.java:23897)
	at org.eclipse.n4js.ide.contentassist.antlr.internal.InternalN4JSParser.rule__Script__Group__1(InternalN4JSParser.java:23887)
	at org.eclipse.n4js.ide.contentassist.antlr.internal.InternalN4JSParser.rule__Script__Group__0(InternalN4JSParser.java:23868)
	... 18 more

```

--- a/plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/editor/contentassist/ContentAssistTokenTypeMapper.java
+++ b/plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/editor/contentassist/ContentAssistTokenTypeMapper.java
@@ -63,7 +63,8 @@ public class ContentAssistTokenTypeMapper {
 		if (type != null) {
 			return type;
 		}
-		throw new IllegalArgumentException(text);
+
+		return InternalN4JSParser.RULE_ANY_OTHER;
 	}
 
 	/**

